[V2][LAYOUT] Authentication screens.

philipbrito opened the pull request:
This PR adds the authentication screens for v2.
Closes #544 

Note:
- I took advantage and added the file that contains the translation for Brazilian Portuguese (default is english), so community, you can do the same and add another translation (like to French and German language :smile: ) .
- The activities and fragments are in the app directory, but it should be in a better place (let's do that after the project architecture definition).

Check out demo: https://youtu.be/ro9L-L3uHQk

TODO:

- [x] Up layout when keyboard is show. 
- [x] Design the Oauth/Social accounts interface.

philipbrito committed:
Update application base theme.

philipbrito committed:
Add styles.

philipbrito committed:
Add strings.

philipbrito committed:
Add brazilian portuguese string translation.

philipbrito committed:
Add styles for v23+.

drawableTint attribute works only for Android API 23+.

philipbrito committed:
Update colors.

philipbrito committed:
Add drawableTint color.

philipbrito committed:
Add dimens.xml

philipbrito committed:
Add values-w820dp/dimens.xml

philipbrito committed:
Add the layout fragment_authentication_two_factor_auth.xml

philipbrito committed:
Add the layout fragment_authentication_sign_up.xml

philipbrito committed:
Add the layout fragment_authentication_server.xml

philipbrito committed:
Add the layout fragment_authentication_log_in.xml

philipbrito committed:
Add the layout activity_authentication.xml

philipbrito committed:
Add drawable/style_edit_text.xml

philipbrito committed:
Add icon drawables.

philipbrito committed:
Add ripple effect.

philipbrito committed:
Add AuthenticationActivity.kt

philipbrito committed:
Add AuthenticationServerFragment.kt

philipbrito committed:
Add activity to the AndroidManifest.xml

philipbrito committed:
Merge branch 'dependencies' into layout/authentication-screens

philipbrito committed:
Add msg_email and update msg_name to msg_name_and_surname strings

philipbrito committed:
Add msg_email and update msg_name to msg_name_and_surname strings (PT-BR).

philipbrito committed:
Update marginTop.

philipbrito committed:
Add text_email EditText and update marginTop and drawableStart.

philipbrito committed:
Format the code

philipbrito committed:
Add ic_email_black_24dp.xml

philipbrito committed:
Add ic_email_black_24dp.xml

philipbrito committed:
Add AuthenticationSignUpFragment.kt

philipbrito committed:
Remove variable

philipbrito committed:
Update tag.

philipbrito committed:
Update circle.yml (Build-Tools 26.0.2)

philipbrito committed:
Add DrawableHelper.kt

philipbrito committed:
Tint EditText drawableStart if Build.VERSION.SDK_INT  <= Build.VERSION_CODES.M (API 23).

philipbrito committed:
Remove wrong package.

philipbrito committed:
Rename file and update drawableStart icon.

philipbrito committed:
Add drawables.

philipbrito committed:
Add new methods.

philipbrito committed:
Update drawableStart icon and update hint.

philipbrito committed:
Add string.

philipbrito committed:
Add AuthenticationTwoFAFragment.kt

philipbrito committed:
Set editText cursor not visible.

philipbrito committed:
Add new user aggrement view.

philipbrito committed:
No need to use two diferent textView to set the color of a part of the text.

philipbrito committed:
Remove attribute.

philipbrito committed:
Show the button log in.

philipbrito committed:
Add AuthenticationLoginFragment.kt

philipbrito committed:
Add KeyboardHelper.kt

philipbrito committed:
Add string

philipbrito committed:
Add color

philipbrito committed:
Add dimension

philipbrito committed:
Add drawable

philipbrito committed:
Add initial social accounts view.

philipbrito committed:
Add hard keyboard listener and EditTexts listeners to hide/show the social accounts when the user tap the EditText views/keyboard is shown

philipbrito committed:
Check if all EditText's text are null or blank before showing the social accounts and sign up msg view.

philipbrito committed:
Better code formattting

philipbrito committed:
Add drawables.

philipbrito committed:
Add strings (content description messages),

philipbrito committed:
Just code formmating.

philipbrito committed:
Add AuthenticationActivity screen orientation (portrait).

philipbrito committed:
Just add the AuthenticationLoginFragment to the activity instead of AuthenticationServerFragment.

philipbrito committed:
Add social accounts interface.

philipbrito committed:
Remove whiteline.

philipbrito committed:
Add useful methods to handle the social accounts interface.

philipbrito committed:
Add LayoutHelper class.

philipbrito committed:
Workaround to adjust the layout when in full screen mode.

leonardoaramaki commented on dependencies.gradle:
```diff
@@ -5,7 +5,7 @@ ext {
         java              : JavaVersion.VERSION_1_8,
         compileSdk        : 26,
```
We should change compileSdk, targetSdk to 27 and buildTools as well as support to 27.0.0 so we can use the latest apis.

leonardoaramaki committed:
Merge branch 'develop-2.x' into layout/authentication-screens

leonardoaramaki committed:
Merge branch 'develop-2.x' into layout/authentication-screens

philipbrito committed:
Merge branch 'dependencies' into layout/authentication-screens

philipbrito committed:
Merge branch 'layout/authentication-screens' of github.com:filipedelimabrito/Rocket.Chat.Android into layout/authentication-screens

philipbrito committed:
Remove commented code.

philipbrito committed:
Update string.

philipbrito committed:
Add try-catch block for safe casting.

philipbrito committed:
Update circle.yml

Just commenting the google-services.json and api_key_strings.xml which are unusable at this moment.

philipbrito committed:
Remove white line.
